tests/test_cognito.py

```diff
@@ -28,3 +28,15 @@ def test_query_user_pool(self):
         self.assertEqual(
             sorted([n['Name'] for n in resources]),
             ['c7nusers', 'origin_userpool_MOBILEHUB_1667653900'])
+
+    def test_delete_user_pool(self):
+        factory = self.replay_flight_data('test_cognito-user-pool_delete')
+        p = self.load_policy({
+            'name': 'delete-users',
+            'resource': 'user-pool',
+            'filters': [{'Name': 'test-delete-user-pool'}],
+            'actions': ['delete']
+        }, session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['Name'], 'test-delete-user-pool')
```

Comment:
to assert the deletion you need to create a cognito client and verify the named pool is no longer extant.

Reply:
Now asserted via client